src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -182,22 +184,28 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
     public List<String> getSectionsNameForCourse(String courseId) throws EntityDoesNotExistException {
+        return getSectionsNameForCourse(courseId, true);
+    }
 
-        verifyCourseIsPresent(courseId);
-        
+    private List<String> getSectionsNameForCourse(
+        String courseId, boolean checkExists) throws EntityDoesNotExistException {
+        if (checkExists) {
+            verifyCourseIsPresent(courseId);    
```

Comment:
I have created the overload methods for now. I think we can create another issue to change other functions to use the one with `CourseAttributes` object.